type-definitions/immutable.js.flow

```diff
@@ -561,9 +561,78 @@ declare module 'immutable' {
     flatten(shallow?: boolean): /*this*/Map<any,any>;
   }
 
-  // OrderedMaps have nothing that Maps do not have. We do not need to override constructor & other statics
-  declare class OrderedMap<K,V> extends Map<K,V> {
+  declare class OrderedMap<K,V> extends KeyedCollection<K,V> {
```

Comment:
Unfortunately the updater function is still problematic. I can work around it by changing the update signature from

```
update<K_,V_>(updater: (value: Map<K,V>) => Map<K_,V_>): Map<K_,V_>;
```

to

```
update<K_,V_>(updater: (value: any) => any): Map<K_,V_>;
```

but I'm not sure how else to fix it. The error being generated is as follows:

```
inconsistent use of library definitions
571:     update<K_,V_>(updater: (value: OrderedMap<K,V>) => OrderedMap<K_,V_>): OrderedMap<K_,V_>;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ intersection. This type is incompatible with
500:     update<K_,V_>(updater: (value: Map<K,V>) => Map<K_,V_>): Map<K_,V_>;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ polymorphic type: function type
  Member 1:
  571:     update<K_,V_>(updater: (value: OrderedMap<K,V>) => OrderedMap<K_,V_>): OrderedMap<K_,V_>;
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ polymorphic type: function type
  Error:
  inconsistent use of library definitions
  500:     update<K_,V_>(updater: (value: Map<K,V>) => Map<K_,V_>): Map<K_,V_>;
                                                       ^^^^^^^^^^ Map. This type is incompatible with
  571:     update<K_,V_>(updater: (value: OrderedMap<K,V>) => OrderedMap<K_,V_>): OrderedMap<K_,V_>;
                                                              ^^^^^^^^^^^^^^^^^ OrderedMap
  Member 2:
  572:     update<V_>(key: K, updater: (value: V) => V_): OrderedMap<K,V|V_>;
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ polymorphic type: function type
  Error:
  inconsistent use of library definitions
  500:     update<K_,V_>(updater: (value: Map<K,V>) => Map<K_,V_>): Map<K_,V_>;
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ undefined (too few arguments, expected default/rest parameters). This type is incompatible with
  572:     update<V_>(key: K, updater: (value: V) => V_): OrderedMap<K,V|V_>;
                                       ^^^^^^^^^^^^^^^^ function type
  Member 3:
  573:     update<V_>(key: K, notSetValue: V_, updater: (value: V) => V_): OrderedMap<K,V|V_>;
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ polymorphic type: function type
  Error:
  inconsistent use of library definitions
  500:     update<K_,V_>(updater: (value: Map<K,V>) => Map<K_,V_>): Map<K_,V_>;
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ undefined (too few arguments, expected default/rest parameters). This type is incompatible with
  573:     update<V_>(key: K, notSetValue: V_, updater: (value: V) => V_): OrderedMap<K,V|V_>;
                                                        ^^^^^^^^^^^^^^^^ function type
```